src/Data/Array/Polarized/Push.hs

```diff
@@ -39,7 +39,8 @@ import qualified Prelude
 -- | Push arrays are un-allocated finished arrays. These are finished
 -- computations passed along or enlarged until we are ready to allocate.
 data Array a where
-  Array :: (forall m. Monoid m => (a -> m) -> m) %1 -> Array a
+  -- TODO: do we really need both 'Prelude.Monoid' and 'Monoid' as constraints?
```

Comment:
This seems to be the only drawback from breaking the relationship between `Data.Monoid.Monoid` and `Data.Monoid.Linear.Monoid`

Reply:
You don't need to do that, I think. In fact, I think that we can use `Prelude.Monoid` only, here, as the elements are presumed to be unrestricted.

Reply:
In fact, we need a linear `Monoid` constraint here, or else we break linear guarantees in `cons` and `snoc`. But I found a workaround.

Reply:
Indeed, my bad.